src/rt/lifetime.d

```diff
@@ -2873,3 +2913,18 @@ unittest
         s.thisptr = &s;
     }
 }
+
+// alignment test
+debug(SENTINEL) {} // block sizes and alignment in SENTINEL mode are not normal.
+else unittest
+{
+    import core.memory;
+    static struct S {
+        align(64) int x;
+    }
+
+    auto arr = new S[1];
+    auto info = GC.query(arr.ptr);
+    assert(cast(size_t)arr.ptr % 64 == 0);
+    assert(info.size % 64 == 0);
```

Comment:
FYI, when I did not turn off this test for `debug(SENTINEL)`, arr.ptr was not aligned to 64 bytes, and info.size was 65!

Reply:
Yes, even 16 byte alignment is not guaranteed with the SENTINEL code IIRC.